demisto_sdk/commands/common/git_util.py

```diff
@@ -0,0 +1,319 @@
+
+    def _get_untracked_files(self, requested_status: str) -> set:
+        """return all untracked files of the given requested status.
+        Args:
+            requested_status (str): M, A, R, D - the git status to return
+        Returns:
+            Set: of path strings which include the untracked files of a certain status.
+        """
+        git_status = self.repo.git.status('--short', '-u').split('\n')
+
+        # in case there are no local changes - return
+        if git_status == ['']:
+            return set()
+
+        extracted_paths = set()
+        for line in git_status:
+            line = line.strip()
+            file_status = line.split()[0].upper() if not line.startswith('?') else 'A'
+            if file_status == requested_status:
+                if requested_status == 'R':
+                    extracted_paths.add((Path(line.split()[-2]), Path(line.split()[-1])))
+                else:
+                    extracted_paths.add(Path(line.split()[-1]))  # type: ignore
+
+        return extracted_paths
+
+    def _get_all_changed_files(self, prev_ver: str) -> Set[Path]:
+        """Get all the files changed in the current branch without status distinction.
+        Args:
+            prev_ver (str): The base branch against which the comparison is made.
+        Returns:
+            Set: of Paths to files changed in the current branch.
+        """
+        remote, branch = self._handle_prev_ver(prev_ver)
+
+        return {Path(os.path.join(item)) for item
+                in self.repo.git.diff('--name-only',
+                                      f'{remote}/{branch}...{self.repo.active_branch}').split('\n')}
+
+    def _only_last_commit(self, prev_ver: str, requested_status: str) -> Set:
+        """Get all the files that were changed in the last commit of a given type when checking a branch against itself.
+        Args:
+            prev_ver (str): The base branch against which the comparison is made.
+            requested_status (str): M, A, R, D - the git status to return
+        Returns:
+            Set: of Paths to files changed in the the last commit or an empty set if not
+            running on master against master.
+        """
+        # when checking branch against itself only return the last commit.
+        if self.get_current_working_branch() == prev_ver:
+            try:
+                if requested_status != 'R':
+                    return {Path(os.path.join(item.a_path)) for item in
+                            self.repo.commit('HEAD~1').diff().iter_change_type(requested_status)}
+                else:
+                    return {(Path(item.a_path), Path(item.b_path)) for item in
+                            self.repo.commit('HEAD~1').diff().iter_change_type(requested_status)}
+            except gitdb.exc.BadName:
+                # in case no last commit exists - just pass
+                pass
+        return set()
+
+    def check_if_remote_exists(self, remote):
```

Comment:
should not it be a closure func inside `_handle_prev_ver`?

Reply:
no, this check is used outside of this internal use as well in order to check if a given remote exists if not I print an error to the user and revert to the default remote.